src/components/Table/index.tsx

```diff
@@ -81,77 +134,95 @@ export const Table = <TData, TValue>({
     }
   }, [table, onRowSelected])
 
-  const TableBodyElement = (): React.ReactNode => {
-    if (table.getRowModel().rows?.length === 0) {
+  // Needed for the useMemo to refresh when new data is set in the table
+  // Defer to previous table and loading state until ready, reduces flickering
+  const deferredRows = useDeferredValue(table.getRowModel().rows)
+  const deferredLoading = useDeferredValue(loading)
+
+  const TableBodyElement = useMemo(() => {
+    if (deferredLoading || !deferredRows?.length) {
       return (
         <TableRow>
           <TableCell
             colSpan={columns.length}
-            className="h-24 text-center"
+            className='h-24 text-center'
           >
-            {loading ? 'Loading...' : 'No results.'}
+            {deferredLoading
+              ? 'Laddar...'
+              : 'Inga planeringar funna.'
+            }
           </TableCell>
         </TableRow>
       )
     }
 
-    return table.getRowModel().rows.map((row) => (
-      <TableRow
-        key={`calendar/${row.id}`}
-        className='cursor-default'
-        data-state={row.getIsSelected() && 'selected'}
-        onClick={<T extends HTMLElement>(event: MouseEvent<T>) => {
-          // @ts-expect-error unknown
-          if (!event.nativeEvent.target.dataset.rowAction) {
-            if (!onRowSelected) {
-              return
-            }
+    return deferredRows.map((row) => {
+      return (
+        <React.Fragment key={row.id}>
+          <TableRow className='sticky top-0 bg-muted'>
+            <TableCell colSpan={columns.length} className='pl-6'>
+              <div className='flex justify-between items-center flex-wrap'>
+                <div className='flex items-center space-x-2'>
+                  <span className='font-thin text-muted-foreground'>Nyhetsvärde</span>
+                  <span className='inline-flex items-center justify-center w-6 h-6 bg-background rounded-full ring-1 ring-gray-300 shadow-md'>
```

Comment:
Can use `size-6` instead of `h-6 w-6`. Also suggesting to make it smaller and removing shadow (see screenshot in PR comment).